src/Promise.php

```diff
@@ -38,6 +68,24 @@ public function run(): void
      */
     public function resolve(): mixed
     {
-        return $this->result->get();
+        try {
+            $result = $this->result->get();
+
+            if ($thenResult = ($this->then)($result)) {
```

Comment:
this would not actually work, because sometimes, the `thenREsult` is actually 'null` or `false`.